api/src/pcapi/alembic/versions/20230405T145827_49ed95a60b07_set_user_public_name_to_nullable.py

```diff
@@ -0,0 +1,20 @@
+"""set_user_public_name_to_nullable
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# pre/post deployment: pre
+# revision identifiers, used by Alembic.
+revision = "49ed95a60b07"
+down_revision = "66c2977b3cfe"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.alter_column("user", "publicName", existing_type=sa.VARCHAR(length=255), nullable=True)
+
+
+def downgrade() -> None:
+    op.alter_column("user", "publicName", existing_type=sa.VARCHAR(length=255), nullable=False)
```

Comment:
```suggestion
    # Do not restore NOT NULL constraint. It will cause downtime on such a large and
    # frequently modified table.
    pass
```

Il est peu probable qu'on "downgrade" cette migration. Et comme il est encore moins probable qu'on se dise que, finalement, on a envie de conserver ce champ et cette contrainte, je pense que ne rien faire dans le downgrade est raisonnable.